can-simple-observable.js

```diff
@@ -56,22 +57,121 @@ SimpleObservable.prototype = {
 		queues.enqueueByQueue(this.handlers.getNode([]), this, [value, old], function(){
 			return {};
 		});
+		//!steal-remove-start
+		callLogHandlers.call(this, this.logHandlers, old, value);
+		//!steal-remove-end
 	},
 	// .on( handler(newValue,oldValue), queue="mutate")
 	on: function(handler, queue){
 		this.handlers.add([queue|| "mutate", handler]);
 	},
 	off: function(handler, queue){
 		this.handlers.delete([queue|| "mutate", handler]);
+	},
+
+	//!steal-remove-start
+	_log: function(name, handler) {
+		return this.logHandlers.push({
+			name: name,
+			handler: handler || function(name, prev) {
+				console.log(name + " (was <" + prev + ">)");
+			}
+		});
+	},
+	//!steal-remove-end
+
+	/**
+	 * Turns on logging of the observable changes to the browser console
+	 *
+	 * Usage:
+	 *
+	 * ```
+	 *   var obs = new Observable("one");
+	 *
+	 *   obs.log();
+	 *
+	 *   // an optional name can be provided to be shown in the log
+	 *   obs.log("myObservable");
+	 *
+	 *  // a log handler can be provided to customize the console output
+	 *  obs.log(function(name, previous, current) {
+	 *    console.table([{ name, previous, current }]);
+	 *  });
+	 * ```
+	 *
+	 * Note: `.log` is a no-op in production
+	 *
+	 * @param {string?} name - Name shown in the log, defaults to the result of
+	 *                         calling canReflect.getName on the observable
+	 * @param {function?} handler - Optional function that receives the name, the
+	 *                              previous and current value; use it to customize
+	 *                              the console output.
+	 */
+	log: function(name, handler) {
+		//!steal-remove-start
+		if (arguments.length === 1) {
+			name = typeof name === "string" ? name : null;
+			handler =  typeof name === "function" ? name : null;
+		}
+		return this._log(name, handler);
+		//!steal-remove-end
+	},
+
+	/**
+	 * Turns off logging, pass name if `.log` was provided one.
+	 *
+	 * E.g:
+	 *
+	 * ```
+	 * var obs = new Observable("one");
+	 * obs.log("myObservable");
+	 *
+	 * ...
+	 *
+	 * // turn off logging
+	 * obs.offLog("myObservable");
+	 * ```
+	 *
+	 * @param {string?} name - The optional name passed to `.log`
+	 */
+	offLog: function(name) {
+		//!steal-remove-start
+		var handlers = this.logHandlers;
+		for (var i = 0; i < handlers.length; i += 1) {
+			if (handlers[i].name === name) {
+				handlers.splice(i, 1);
+			}
+		}
+		//!steal-remove-end
 	}
 };
 
+/**
```

Comment:
don't document internal functions like this ... we don't want them showing up in canjs' docs.

Reply:
Using line comments (`// ...`) instead of JSDoc comments (`/**...`) will exclude them from the output of bit-docs.  Then you can still retain documentation for other developers while not making these functions part of the published API.